packages/server/src/fhir/storage.ts

```diff
@@ -57,18 +57,22 @@ class FileSystemStorage implements BinaryStorage {
     binary: Binary,
     filename: string | undefined,
     contentType: string | undefined,
-    stream: Readable | NodeJS.ReadableStream
+    input: Readable | NodeJS.ReadableStream
   ): Promise<void> {
     checkFileMetadata(filename, contentType);
     const dir = this.getDir(binary);
     if (!existsSync(dir)) {
       mkdirSync(dir);
     }
     const writeStream = createWriteStream(this.getPath(binary), { flags: 'w' });
-    stream.pipe(writeStream);
     return new Promise((resolve, reject) => {
-      writeStream.on('close', resolve);
-      writeStream.on('error', reject);
+      pipeline(input, writeStream, (err) => {
```

Comment:
Out of curiosity, what is the reason behind this?

Reply:
Nothing major, but I preferred this style for two reasons:
1. It co-locates the pipe operation and the handling of ultimate completion or error neatly
2. It abstracted away the details of which events were necessary to listen to, which I worry about with streams because I don't know them super well